Fixes #29331 - use patternfly v4 fonts

sharvit opened the pull request:
Use pf4 fonts (RedhatText) and link colors (blue).

# Examples

![screenshot-centos7-luna-devel sharvit-fedorabook-t480s example com-2020 03 15-14_10_42](https://user-images.githubusercontent.com/1262502/76701106-c1d8f380-66c6-11ea-911a-51ddc1e284be.png)

![screenshot-centos7-luna-devel sharvit-fedorabook-t480s example com-2020 03 15-14_15_58 (1)](https://user-images.githubusercontent.com/1262502/76701182-7d018c80-66c7-11ea-9b48-7926bb5e4f64.png)

![screenshot-centos7-luna-devel sharvit-fedorabook-t480s example com-2020 03 15-14_15_58](https://user-images.githubusercontent.com/1262502/76701184-7ecb5000-66c7-11ea-8d90-fb3fe6eaf649.png)

sharvit committed:
Fixes #29331 - use patternfly v4 fonts

amirfefer commented on app/assets/stylesheets/navigation.scss:
```diff
@@ -44,6 +44,10 @@
   span {
```
does it make sense moving navigation styles to webpack?

sharvit replied:
No, we aim to replace the navigation component later.
Anyway, not in this PR please.

amirfefer reviewed (commented):
Overall I'm :+1: with this change, though it isn't dramatically noticeable, it's a good strart

amirfefer commented on package.json:
```diff
@@ -20,7 +20,7 @@
   },
   "dependencies": {
-    "@theforeman/vendor": "^4.0.7",
+    "@theforeman/vendor": "^4.2.0",
```
do we need packaging ack here?

johnpmitsch commented on package.json:
```diff
@@ -30,11 +30,11 @@
+    "@theforeman/eslint-plugin-foreman": "^4.2.0",
+    "@theforeman/stories": "^4.2.0",
+    "@theforeman/test": "^4.2.0",
+    "@theforeman/vendor-dev": "^4.2.0",
```
@sharvit I opened up https://github.com/theforeman/foreman/pull/7590 - but I didn't see the versions were updated here. Should these be `^5.0.0` now that that is released?

johnpmitsch commented:
@sharvit  Thanks for the hard work getting pf4 into Foreman. The package bumps and fonts are needed for https://github.com/Katello/katello/pull/8644

Is there anything I can do to help move this PR along?

sharvit commented:
Thanks @johnpmitsch and @amirfefer.

I ended up doing a reset to the commits in v5.0.0 and deprecated that version.
Sorry for the confusion about it.

packaging-pr:
https://github.com/theforeman/foreman-packaging/pull/5106

tbrisker reviewed (commented):
The secondary menu item font size seems very small with this change.
New hosts dashboard widget overflows if there are enough new hosts, i assume the same is true of other widgets:
![image](https://user-images.githubusercontent.com/1540531/80371432-94bd5a80-889a-11ea-8258-9cf388a6a6db.png)

sharvit commented:
Thanks @tbrisker, I changed the base font-size from `--pf-global--FontSize--sm` to `--pf-global--FontSize--xs` while the recommended is `--pf-global--FontSize--md`.

Now it should look more similar to what we had before so those layout issues are now fixed. From the other side, I see it as a technical debt because pf4 encourages using bigger fonts and fewer components in one screen so the user would have more focus.

IMO we can start with the `xs` font size and over time increase it to `md`.
We can set the font to `sm` after branching and try to address layout issues until the next branching.

tbrisker commented:
> Thanks @tbrisker, I changed the base font-size from `--pf-global--FontSize--sm` to `--pf-global--FontSize--xs` while the recommended is `--pf-global--FontSize--md`.
> 
> Now it should look more similar to what we had before so those layout issues are now fixed. From the other side, I see it as a technical debt because pf4 encourages using bigger fonts and fewer components in one screen so the user would have more focus.
> 
> IMO we can start with the `xs` font size and over time increase it to `md`.
> We can set the font to `sm` after branching and try to address layout issues until the next branching.

Looks like at least part of the overflow is due to the header breaking up (operating system), would it make sense to modify column widths instead so it all fits and/or reduce the number of lines displayed in the widget?

tbrisker commented:
Also, packaging has been merged, this needs a rebase

sharvit commented:
> Looks like at least part of the overflow is due to the header breaking up (operating system), would it make sense to modify column widths instead so it all fits and/or reduce the number of lines displayed in the widget?

Yes, modified the os column and set the tables to have pf4 line-height and it looks fine with the `sm` font size.

tbrisker commented:
Looks like there's still some font sizes that are off, mostly the navbar menus are very small but also the action buttons seem a bit smaller than the rest of the row:
![Screenshot from 2020-05-26 15-52-20](https://user-images.githubusercontent.com/1540531/82903133-2f2bbf00-9f69-11ea-93f3-abda8d0ef881.png)

sharvit commented:
Thanks @tbrisker, I did some small adjustments so it would be similar to what we have now.

tbrisker commented:
Please rebase @sharvit

sharvit commented:
Rebased

tbrisker reviewed (approved):
Tested, looks good! The menu items are still a tad bit small, but let's fix that and any additional issues in following PRs.
Thanks @sharvit !
